src/main/scala/mesosphere/marathon/core/externalvolume/impl/providers/DVDIProvider.scala

```diff
@@ -111,28 +113,45 @@ private[impl] object DVDIProviderValidations extends ExternalVolumeValidations {
   import mesosphere.marathon.api.v2.Validation._
   import DVDIProvider._
 
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
   // group-level validation for DVDI volumes: the same volume name may only be referenced by a single
   // task instance across the entire cluster.
   override lazy val rootGroup = new Validator[Group] {
     override def apply(g: Group): Result = {
-      val appsByVolume =
+      val appsByVolume: Map[String, Set[PathId]] =
         g.transitiveApps
           .flatMap { app => namesOfMatchingVolumes(app).map(_ -> app.id) }
           .groupBy { case (volumeName, _) => volumeName }
           .mapValues(_.map { case (volumeName, appId) => appId })
 
-      val groupViolations = g.apps.flatMap { app =>
```

Comment:
The `g.apps` instead of `g.transitiveApps` prevented validation of apps in nested groups.